src/query/interpret/awesome_memgraph_functions.cpp

```diff
@@ -1383,6 +1397,281 @@ TypedValue DateTime(const TypedValue *args, int64_t nargs, const FunctionContext
+    }
+    throw QueryRuntimeException("Argument {} is not numeric.", arg_name);
+  };
+
+  auto [x, from_longitude] = std::invoke([&]() -> std::pair<double, bool> {
+    auto it_x = input.find("x");
+    if (it_x != input.end()) {
+      return {numeric_as_double(it_x->second, "longitude/x"), false};
+    }
+    auto it_longitude = input.find("longitude");
+    if (it_longitude != input.end()) {
+      return {numeric_as_double(it_longitude->second, "longitude/x"), true};
+    }
+    throw QueryRuntimeException("Argument longitude/x is missing.");
+  });
+
+  auto [y, from_latitude] = std::invoke([&]() -> std::pair<double, bool> {
+    auto it_y = input.find("y");
+    if (it_y != input.end()) {
+      return {numeric_as_double(it_y->second, "latitude/y"), false};
+    }
+    auto it_latitude = input.find("latitude");
+    if (it_latitude != input.end()) {
+      return {numeric_as_double(it_latitude->second, "latitude/y"), true};
+    }
+    throw QueryRuntimeException("Argument latitude/y is missing.");
+  });
+
+  if (from_longitude != from_latitude) {
+    throw QueryRuntimeException("Use either x, y, z or longitude, latitude, height.");
+  }
+
+  using z_type = std::optional<std::pair<double, bool>>;
+  auto z_opt = std::invoke([&]() -> z_type {
+    auto it_z = input.find("z");
+    if (it_z != input.end()) {
+      return z_type{std::in_place, numeric_as_double(it_z->second, "height/z"), false};
+    }
+    auto it_height = input.find("height");
+    if (it_height != input.end()) {
+      return z_type{std::in_place, numeric_as_double(it_height->second, "height/z"), true};
+    }
+    return std::nullopt;
+  });
+
+  auto crs = std::invoke([&]() -> std::optional<std::string_view> {
+    auto value = input.find("crs");
+    if (value == input.end() || !value->second.IsString()) {
+      return std::nullopt;
+    }
+    return value->second.ValueString();
+  });
+
+  auto srid = std::invoke([&]() -> std::optional<int> {
+    auto value = input.find("srid");
+    if (value == input.end() || !value->second.IsInt()) {
+      return std::nullopt;
+    }
+    return value->second.ValueInt();
+  });
+
+  if (crs.has_value() && srid.has_value()) {
+    throw QueryRuntimeException("Cannot specify both CRS and SRID.");
+  }
+
+  std::optional<storage::CoordinateReferenceSystem> mg_crs;
+  if (crs.has_value()) {
+    mg_crs = storage::StringToCrs(*crs);
+    if (!mg_crs.has_value()) {
+      throw QueryRuntimeException("Invalid CRS.");
+    }
+  } else if (srid.has_value()) {
+    mg_crs = storage::SridToCrs(static_cast<storage::Srid>(*srid));
+    if (!mg_crs.has_value()) {
+      throw QueryRuntimeException("Invalid SRID.");
+    }
+  }
+
+  using CRS = storage::CoordinateReferenceSystem;
+  if (!mg_crs.has_value()) {
+    auto is_wgs = (from_longitude || from_latitude);
+    if (!z_opt.has_value()) {
+      mg_crs = is_wgs ? CRS::WGS84_2d : CRS::Cartesian_2d;
+    } else {
+      mg_crs = is_wgs ? CRS::WGS84_3d : CRS::Cartesian_3d;
+    }
+  }
```

Comment:
Should `RETURN point({longitude:1, latitude:2, srid:7205}).srid;` be a valid query? In this case we combine longitude, latitude and `srid` value of Cartesian, and from lines 1512 to 1518 it seems that if you define longitude/latitude you should use crs which is `wgs84_2d` or  `wgs84_3d`?